_playwright-tests/test-utils/src/helpers/poll.ts

```diff
@@ -16,3 +24,23 @@ export const poll = async (
 
 // sleep in ms, must await e.g. await sleep(num)
 export const sleep = (ms: number) => new Promise((res) => setTimeout(res, ms));
+
+/**
+ * Waits for a repository introspection to complete by polling the repository status.
+ * @param client - Configuration object for API client (optional, uses default if not provided)
+ * @param repoUuid - UUID of the repository to check
+ * @param interval - Polling interval in seconds (default: 10)
+ */
+export const waitForRepositoryIntrospection = async (
+  client: Configuration | undefined,
```

Comment:
this shouldn't be optional, not sure what you mean by it actually using `default` if not provided 🤔 
the client is being used in this helper so it is required, please change that 🙏🏼